src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -586,10 +586,11 @@ private void OnEnvDTEProjectRemoved(EnvDTE.Project envDTEProject)
             // This is a solution event. Should be on the UI thread
             ThreadHelper.ThrowIfNotOnUIThread();
 
-            RemoveVsProjectAdapterFromCache(envDTEProject.FullName);
             NuGetProject nuGetProject;
             _projectSystemCache.TryGetNuGetProject(envDTEProject.Name, out nuGetProject);
 
+            RemoveVsProjectAdapterFromCache(envDTEProject.FullName);
```

Comment:
Moving this statement fixed a bug where the `NuGetProjectRemoved` event would fire with a `null` `NuGetProject`.  This is because the project would first be removed from the project cache before we'd attempt to get it from the project cache to include in the event args.